realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -784,6 +785,82 @@ private Scanner getFullStringScanner(InputStream in) {
     }
 
     /**
+     * Bulk insert of a list of RealmObjects, this is faster than {@link #copyToRealm(Iterable)} since it
+     * doesn't return the inserted elements, and doesn't perform checks (threads, parameters etc.).
+     * <p>
+     * Please note, copying an object will copy all field values. Any unset field in the object and child objects will be
+     * set to their default value if not provided.
+     *
+     * @param objects RealmObjects to insert.
+     */
+    public void insertToRealm(Collection<? extends RealmModel> objects) {
+        Iterator<? extends RealmModel> iterator = objects.iterator();
+        RealmModel object = null;
+        Map<RealmModel, Long> cache = new IdentityHashMap<RealmModel, Long>(objects.size());
+        if (iterator.hasNext()) {
+            // access the first element to figure out the type
+            object = iterator.next();
+            configuration.getSchemaMediator().insertToRealm(this, object, cache);
+        }
+
+        if (iterator.hasNext()) {
+            configuration.getSchemaMediator().insertToRealm(this, object, iterator, cache);
+        }
+    }
+
+    /**
+     * Insert of a RealmObject, this is faster than {@link #copyToRealm(RealmModel)} since it
+     * doesn't return the inserted elements, and doesn't perform checks (threads, parameters etc.).
+     * <p>
+     * Please note, copying an object will copy all field values. Any unset field in the object and child objects will be
+     * set to their default value if not provided.
+     *
+     * @param object RealmObjects to insert.
+     */
+    public void insertToRealm(RealmModel object) {
+        Map<RealmModel, Long> cache = new IdentityHashMap<RealmModel, Long>();
+        configuration.getSchemaMediator().insertToRealm(this, object, cache);
+    }
+
+    /**
+     * Bulk insert or update of a list of RealmObjects, this is faster than {@link #copyToRealmOrUpdate(Iterable)} since it
+     * doesn't return the inserted elements, and doesn't perform checks (threads, parameters etc.).
+     * <p>
+     * Please note, copying an object will copy all field values. Any unset field in the object and child objects will be
+     * set to their default value if not provided.
+     *
+     * @param objects RealmObjects to insert.
+     */
+    public void insertOrUpdateToRealm(Collection<? extends RealmModel> objects) {
```

Comment:
IMO our API is already a bit to verbose in places. No need to go further down that route 😄 . `insertRealmObject(RealmModel obj)` and `insertRealmCollection(Collection<RealmModel> collection)` basically say the same thing twice (name + parameter).

If you look at the combine method description I am strongly leaning towards:
- `insert(RealmModel obj)`
- `insertOrUpdate(RealmModel obj)`
- `insert(Collection<RealmModel> collection)`
- `insertOrUpdate(Collection<RealmModel> collection)`

They are descriptive without being verbose and sets the apart from `copy<To/From>Realm` also has benefits I think. Also the name is similar to what you have in SQLite and other ORM's.